tiledb/sm/filesystem/s3.h

```diff
@@ -80,14 +81,174 @@
 #include <vector>
 
 using namespace tiledb::common;
+using tiledb::common::filesystem::directory_entry;
+
+namespace tiledb::sm {
+
+struct S3Parameters {
+  S3Parameters() = delete;
+
+  S3Parameters(const Config& config)
+      : logging_level_(config.get<std::string>("vfs.s3.logging_level").value())
```

Comment:
You can use `config.get<std::string>("vfs.s3.logging_level", Config::must_find)` and remove the `.value()` at the end. This will fail with a detailed error message if the option is not found.